aws/templates/id/id_cognito.ftl

```diff
@@ -1,30 +1,35 @@
 [#-- Cognito UserPool --]
 
 [#assign USERPOOL_RESOURCE_TYPE = "userpool"]
-[#assign USERPOOL_CLIENT_RESOURCE_TYPE = "userpoolclient" ]
-[#assign IDENTITYPOOL_RESOURCE_TYPE = "identitypool" ]
+[#assign USERPOOL_CLIENT_RESOURCE_TYPE = "userpoolclient"]
+[#assign IDENTITYPOOL_RESOURCE_TYPE = "identitypool"]
 
-[#function formatUserPoolId tier component extensions...]
+[#assign USERPOOL_COMPONENT_TYPE = "userpool"]
+
+[#function formatUserPoolId occurrence extensions...]
```

Comment:
these could simplify further to

[#return formatResourceId(USERPOOL_RESOURCE_TYPE,occurrence.Core.Id, extensions)

since Core.Id is effectively tier, component and id extensions